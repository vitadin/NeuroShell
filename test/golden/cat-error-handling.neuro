%% Cat command graceful degradation test
\set[_echo_command="true"]
%% Tests graceful handling of invalid option values

%% Test invalid option values (should degrade gracefully)
\cat[path=test/fixtures/simple.txt, silent=invalid_bool]
\get[_output]

%% Test invalid lines value (should degrade gracefully)
\cat[path=test/fixtures/simple.txt, lines=invalid_number]
\get[_output]

%% Test invalid start value (should degrade gracefully)
\cat[path=test/fixtures/simple.txt, start=invalid_number]
\get[_output]

%% Test negative lines value (should degrade gracefully)
\cat[path=test/fixtures/simple.txt, lines=-5]
\get[_output]

%% Test very large start value (should result in empty output)
\cat[path=test/fixtures/simple.txt, start=9999, to=large_start]
\get[large_start]

%% Test combined invalid options (should still work with valid fallbacks)
\cat[path=test/fixtures/simple.txt, silent=invalid, lines=abc, start=xyz, to=combined_invalid]
\get[combined_invalid]

%% Test zero and negative values
\cat[path=test/fixtures/simple.txt, lines=0, start=0, to=zero_values]
\get[zero_values]

%% Test string values for numeric options
\cat[path=test/fixtures/simple.txt, lines=hello, start=world, to=string_values]
\get[string_values]