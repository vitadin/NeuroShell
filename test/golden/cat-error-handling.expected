Setting _echo_command = true
%%> "\\cat[path=test/fixtures/simple.txt, silent=invalid_bool]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[_output]"
_output = This is a simple text file.
It has multiple lines for testing.   
Line 3 contains some content.        
Final line of the file.              
%%> "\\cat[path=test/fixtures/simple.txt, lines=invalid_number]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[_output]"
_output = This is a simple text file.
It has multiple lines for testing.   
Line 3 contains some content.        
Final line of the file.              
%%> "\\cat[path=test/fixtures/simple.txt, start=invalid_number]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[_output]"
_output = This is a simple text file.
It has multiple lines for testing.   
Line 3 contains some content.        
Final line of the file.              
%%> "\\cat[path=test/fixtures/simple.txt, lines=-5]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[_output]"
_output = This is a simple text file.
It has multiple lines for testing.   
Line 3 contains some content.        
Final line of the file.              
%%> "\\cat[path=test/fixtures/simple.txt, start=9999, to=large_start]"
%%> "\\get[large_start]"
large_start = 
%%> "\\cat[path=test/fixtures/simple.txt, silent=invalid, lines=abc, start=xyz, to=combined_invalid]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[combined_invalid]"
combined_invalid = This is a simple text file.
It has multiple lines for testing.            
Line 3 contains some content.                 
Final line of the file.                       
%%> "\\cat[path=test/fixtures/simple.txt, lines=0, start=0, to=zero_values]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[zero_values]"
zero_values = This is a simple text file.
It has multiple lines for testing.       
Line 3 contains some content.            
Final line of the file.                  
%%> "\\cat[path=test/fixtures/simple.txt, lines=hello, start=world, to=string_values]"
This is a simple text file.
It has multiple lines for testing.
Line 3 contains some content.
Final line of the file.
%%> "\\get[string_values]"
string_values = This is a simple text file.
It has multiple lines for testing.         
Line 3 contains some content.              
Final line of the file.                    